ed_perception/src/perception_plugin_image_recognition.cpp

```diff
@@ -105,16 +108,20 @@ bool PerceptionPluginImageRecognition::srvClassify(ed_perception_msgs::Classify:
 
         for(ed::ImageMask::const_iterator it2 = mask.begin(image.cols); it2 != mask.end(); ++it2)
         {
-            const cv::Point2i p(it2());
+            const cv::Point& p(it2());
             p_min.x = std::min(p_min.x, p.x);
             p_min.y = std::min(p_min.y, p.y);
             p_max.x = std::max(p_max.x, p.x);
             p_max.y = std::max(p_max.y, p.y);
         }
-        cv::Rect roi = cv::Rect(std::min(p_min.x + 5, image.cols),
-                                std::min(p_min.y + 5, image.rows),
-                                std::max(p_max.x - p_min.x - 5, 0),
-                                std::max(p_max.y - p_min.y - 5, 0));
+        
+        int x = std::max(std::min(p_min.x - roi_margin_, image.cols), 0);
```

Comment:
These statements only need a max or a min. We know p_min and p_max lie within the image and margin >0

Reply:
No, margin can also be negative. Therefore `p_min-margin` can be outside the image. So both `min`/`max` are needed.